engine/src/main/java/org/terasology/persistence/typeHandling/Serializer.java

```diff
@@ -146,11 +146,9 @@ public void deserializeOnto(Object target, Map<FieldMetadata<?, ?>, PersistedDat
      * @param check  A check to filter which fields to deserialize
      */
     public void deserializeOnto(Object target, Map<FieldMetadata<?, ?>, PersistedData> values, DeserializationContext context, DeserializeFieldCheck check) {
-        for (Map.Entry<FieldMetadata<?, ?>, PersistedData> field : values.entrySet()) {
-            if (check.shouldDeserialize(classMetadata, field.getKey())) {
-                deserializeOnto(target, field.getKey(), field.getValue(), context);
-            }
-        }
+        values.entrySet().stream().filter(field -> check.shouldDeserialize(classMetadata, field.getKey())).forEach(field -> {
```

Comment:
I'd go with the original version, because I find it easier to read and it's called very often.